server/proxy/grpcproxy/cluster.go

```diff
@@ -113,9 +113,9 @@ func (cp *clusterProxy) monitor(wa endpoints.WatchChannel) {
 			for _, up := range updates {
 				switch up.Op {
 				case endpoints.Add:
-					cp.umap[up.Endpoint.Addr] = up.Endpoint
+					cp.umap[up.Key] = up.Endpoint
 				case endpoints.Delete:
-					delete(cp.umap, up.Endpoint.Addr)
```

Comment:
when op delete, up.Endpoint.Addr is not specified, can not delete the endpoint from map, use up.Key instead

Reply:
looks good, I think using Key is ok because it's already used in resolver.watch https://github.com/etcd-io/etcd/blob/249c0d71d4e106e0d2f5576b9638f2e8bd75ca47/client/v3/naming/resolver/resolver.go#LL97C6-L97C12
Not sure why Endpoint.Addr was used initially